routing/dht/providers/providers.go

```diff
@@ -81,7 +81,7 @@ func NewProviderManager(ctx context.Context, local peer.ID, dstore ds.Datastore)
 const providersKeyPrefix = "/providers/"
 
 func mkProvKey(k key.Key) ds.Key {
-	return ds.NewKey(providersKeyPrefix + hex.EncodeToString([]byte(k)))
+	return ds.NewKey(providersKeyPrefix + base64.StdEncoding.EncodeToString([]byte(k)))
 }
```

Comment:
@whyrusleeping I would use RawURLEncoding.  (1) StdEncoding includes '/' in the Alphabet (see https://tools.ietf.org/html/rfc4648) URLEncoding does not (hence it's name) (2) The RAW form does not include the unnecessary '=' padding character.